composer/utils/remote_uploader.py

```diff
@@ -12,30 +12,49 @@
 import time
 import uuid
 from concurrent.futures import Future, ProcessPoolExecutor
-from typing import List
+from typing import List, Optional
 
-from composer.utils.dist import get_local_rank
+from composer.utils.dist import broadcast_object_list, get_global_rank, get_local_rank
 from composer.utils.file_helpers import (
     maybe_create_object_store_from_uri,
+    parse_uri,
+    validate_credentials,
 )
-from composer.utils.object_store.object_store import ObjectStore, ObjectStoreTransientError
+from composer.utils.object_store.mlflow_object_store import MLFLOW_DBFS_PATH_PREFIX, MLFlowObjectStore
+from composer.utils.object_store.object_store import (
+    ObjectStore,
+    ObjectStoreTransientError,
+)
+from composer.utils.object_store.uc_object_store import UCObjectStore
 from composer.utils.retrying import retry
 
 log = logging.getLogger(__name__)
 
 __all__ = ['RemoteUploader']
 
 
+def _build_dbfs_backend(path: str) -> ObjectStore:
+    if path.startswith(MLFLOW_DBFS_PATH_PREFIX):
+        return MLFlowObjectStore(path=path)
+    UCObjectStore.validate_path(path)
+    return UCObjectStore(path=path)
+
+
 def _upload_file_to_object_store(
     remote_folder: str,
+    is_dbfs: bool,
+    dbfs_path: str,
     remote_file_name: str,
     local_file_path: str,
     overwrite: bool,
     num_attempts: int,
 ) -> int:
-    object_store: ObjectStore = maybe_create_object_store_from_uri(
-        remote_folder,
-    )  # pyright: ignore[reportGeneralTypeIssues]
+    if is_dbfs:
+        object_store: ObjectStore = _build_dbfs_backend(dbfs_path)
+    else:
+        object_store: ObjectStore = maybe_create_object_store_from_uri(
```

Comment:
If `_build_dbfs_backend` is part of `maybe_create_object_store_from_uri` then you don't need is_dbfs and dbfs_path